From: Chad Dupuis <cdupuis@redhat.com>
Date: Tue, 3 Aug 2010 13:50:41 -0400
Subject: [net] netxen: disable on NX_P3_B1 hardware
Message-id: <20100803134757.2239.76461.sendpatchset@localhost.localdomain>
Patchwork-id: 27327
O-Subject: [RHEL 5.6 PATCH 25/44] netxen: The driver doesn't work on NX_P3_B1 so
	cause probe to fail.
Bugzilla: 562937
RH-Acked-by: David S. Miller <davem@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>

Bugzilla
========

562937

Upstream Status
===============

net-2.6 commit id 688328c7ec3cd0dc3b16342aeb045d28012cc955

Description
===========

>From 69fea45ebc3ffc549f9a183427f34440f77c8710 Mon Sep 17 00:00:00 2001
From: Eric W. Biederman <ebiederm@xmission.com>
Date: Wed, 17 Mar 2010 22:24:39 +0000
Subject: [PATCH] netxen: The driver doesn't work on NX_P3_B1 so cause probe to fail.

I haven't been able to get link up on a NX_P3_B1 since 2.6.31.  The
driver complains about a firmware hang instead.  When I asked I was
told rev 0x41 was a preproduction rev.  So disable support in the
driver so no one is surprised the code doesn't work.

Signed-off-by: Eric W. Biederman <ebiederm@aristanetworks.com>
Signed-off-by: David S. Miller <davem@davemloft.net>

diff --git a/drivers/net/netxen/netxen_nic_main.c b/drivers/net/netxen/netxen_nic_main.c
index ff67018..e3b57f5 100644
--- a/drivers/net/netxen/netxen_nic_main.c
+++ b/drivers/net/netxen/netxen_nic_main.c
@@ -1274,8 +1274,8 @@ netxen_nic_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
 
 	pci_read_config_byte(pdev, PCI_REVISION_ID, &revision_id);
 
-	if ((revision_id >= NX_P3_A0) && (revision_id < NX_P3_B1)) {
-		printk(KERN_WARNING "%s: chip revisions between 0x%x-0x%x"
+	if (revision_id >= NX_P3_A0 && revision_id <= NX_P3_B1) {
+		printk(KERN_WARNING "%s: chip revisions between 0x%x-0x%x "
 		       "will not be enabled.\n",
 		       module_name(THIS_MODULE), NX_P3_A0, NX_P3_B1);
 		return -ENODEV;
